common/reflect_resource.go

```diff
@@ -79,21 +79,24 @@ func reflectKind(k reflect.Kind) string {
 	return n
 }
 
-func chooseFieldNameWithAliases(typeField reflect.StructField, aliases map[string]string) string {
+func chooseFieldNameWithAliases(typeField reflect.StructField, parentTypeName string, aliases map[string]map[string]string) string {
```

Comment:
Suggestion: let's make `parentTypeName` type `reflect.Type` rather than `string`. String is too broad as a useful interface, putting burden on the caller to pass the value of the right type.

Reply:
Good point, updated